test/unit/user/implementation/UserSettingsServiceTest.php

```diff
@@ -74,36 +87,126 @@ public function testGet(
+                'expected' => $this->createUserSetting('uiLang', null, 'Europe/Madrid'),
+                'user' => $this->getUserMock('uiLang', null, 'Europe/Madrid'),
+                'isSolarDesignEnabled' => true,
             ],
             'Settings for a user with data language set' => [
-                'expected' => new UserSettings('Europe/Madrid', null, 'defLang'),
-                'user' => $this->getUserMock('', 'defLang', 'Europe/Madrid'),
+                'expected' => $this->createUserSetting(null, 'defLang', 'Europe/Madrid'),
+                'user' => $this->getUserMock(null, 'defLang', 'Europe/Madrid'),
+                'isSolarDesignEnabled' => true,
             ],
             'Settings for a user with UI and data language set' => [
-                'expected' => new UserSettings('Europe/Madrid', 'uiLang', 'defLang'),
+                'expected' => $this->createUserSetting('uiLang', 'defLang', 'Europe/Madrid'),
                 'user' => $this->getUserMock('uiLang', 'defLang', 'Europe/Madrid'),
+                'isSolarDesignEnabled' => true,
+            ],
+            'Settings for a user with interface mode' => [
+                'expected' => $this->createUserSetting(
+                    null,
+                    null,
+                    'Europe/Madrid',
+                    GenerisRdf::PROPERTY_USER_INTERFACE_MODE_ADVANCED
+                ),
+                'user' => $this->getUserMock(
+                    null,
+                    null,
+                    'Europe/Madrid',
+                    GenerisRdf::PROPERTY_USER_INTERFACE_MODE_ADVANCED
+                ),
+                'isSolarDesignEnabled' => true,
+            ],
+            'Settings for a user with interface mode when solar design disabled' => [
+                'expected' => $this->createUserSetting(
+                    null,
+                    null,
+                    'Europe/Madrid'
+                ),
+                'user' => $this->getUserMock(
+                    null,
+                    null,
+                    'Europe/Madrid',
+                    GenerisRdf::PROPERTY_USER_INTERFACE_MODE_ADVANCED
+                ),
+                'isSolarDesignEnabled' => false,
             ],
         ];
     }
 
+    private function createUserSetting(
+        string $uiLanguageUri = null,
+        string $defLangUri = null,
+        string $userTimezone = null,
+        string $userInterfaceMode = null
+    ): UserSettingsInterface {
+        $userSettings = new UserSettings($userTimezone);
+
+        $userSettings->setSetting(UserSettingsInterface::INTERFACE_MODE, $userInterfaceMode);
+        $userSettings->setSetting(UserSettingsInterface::UI_LANGUAGE_CODE, $uiLanguageUri);
+        $userSettings->setSetting(UserSettingsInterface::DATA_LANGUAGE_CODE, $defLangUri);
+        $userSettings->setSetting(UserSettingsInterface::TIMEZONE, $userTimezone);
+
+        return $userSettings;
```

Comment:
chore: Can be simplified with fluent interface

```suggestion
        return $userSettings->setSetting(UserSettingsInterface::INTERFACE_MODE, $userInterfaceMode)
            ->setSetting(UserSettingsInterface::UI_LANGUAGE_CODE, $uiLanguageUri)
            ->setSetting(UserSettingsInterface::DATA_LANGUAGE_CODE, $defLangUri)
            ->setSetting(UserSettingsInterface::TIMEZONE, $userTimezone);
```